carma/launch/localization.launch.py

```diff
@@ -371,6 +373,11 @@ def generate_launch_description():
         map_param_loader_container,
         pcd_map_file_loader_container,
         ndt_matching_container,
-        ekf_localizer_container,
         subsystem_controller
     ]) 
+
+    if simulation_mode is False:
```

Comment:
I'm not sure whether we are actually able to use LaunchArguments directly in python conditional statements like this. I think the ROS2 launch approach for this would be to use IfConditional() instead. To do this, you can add:

```
from launch.conditions import IfCondition
from launch.substitutions import LaunchConfiguration, PythonExpression

    ...

    carma_port_drayage_plugin_container = ComposableNodeContainer(
        condition=IfCondition(PythonExpression(['not ', simulation_mode])), # Nodes in this container will only be launched if 'simulation_mode' evaluates to 'False'
    
    ...
```